sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -269,8 +300,8 @@ def test_error_on_missing_requests_for_sub_estimator(metaestimator):
     # Test that a UnsetMetadataPassedError is raised when the sub-estimator's
     # requests are not set
     if "estimator" not in metaestimator:
-        # This test only makes sense for metaestimators which have a
-        # sub-estimator
+        # If dictionary in METAESTIMATORS doesn't provide a key "estimator":
+        # skip rest of the test
```

Comment:
your new comment is explaining the if condition, i.e. writing something twice, the previous comment was trying to explain why the if condition is there.

Reply:
Indeed, comments are hard but I agree with @adrinjalali. I might add a small example to make it even more explicit but that's it:

```python
        # This test only makes sense for metaestimators which have a
        # sub-estimator, e.g. MyMetaEstimator(estimator=MySubEstimator())
```